crates/net/network/src/transactions/config.rs

```diff
@@ -149,3 +155,124 @@ impl FromStr for TransactionPropagationKind {
+        /// If true, the peer sending this annoucement should be penalized.
+        penalize_peer: bool,
+    },
+}
+
+/// A policy that defines how to handle incoming transaction annoucements,
+/// particularly concerning transaction types and other annoucement metadata.
+pub trait AnnouncementFilteringPolicy: Send + Sync + Unpin + 'static {
+    /// Decides how to handle a transaction announcement based on its type, hash, and size.
+    fn decide_on_announcement(&self, ty: u8, hash: &B256, size: usize) -> AnnouncementAcceptance;
+}
+
+/// An `AnnouncementFilteringPolicy` that enforces strict validation of transaction types.
+#[derive(Debug, Clone, Default)]
+pub struct StrictAnnouncementFilter;
+
+impl AnnouncementFilteringPolicy for StrictAnnouncementFilter {
+    fn decide_on_announcement(&self, ty: u8, hash: &B256, size: usize) -> AnnouncementAcceptance {
+        match TxType::try_from(ty) {
+            Ok(_valid_type) => AnnouncementAcceptance::Accept,
+            Err(_) => {
+                tracing::trace!(target: "net::tx::policy::strict",
+                    %ty,
+                    %size,
+                    %hash,
+                    "Invalid or unrecognized transaction type in announcement. Rejecting entry and recommending peer penalization."
+                );
+                AnnouncementAcceptance::Reject { penalize_peer: true }
+            }
+        }
+    }
+}
+
+/// An `AnnouncementFilteringPolicy` that is more permissive towards unknown transaction types.
+///
+/// Allows for handling of new or future transaction types that a local node may not yet fully
+/// support or recognize.
+#[derive(Debug, Clone, Default)]
+pub struct RelaxedAnnouncementFilter;
+
+impl AnnouncementFilteringPolicy for RelaxedAnnouncementFilter {
+    fn decide_on_announcement(&self, ty: u8, hash: &B256, size: usize) -> AnnouncementAcceptance {
+        match TxType::try_from(ty) {
+            Ok(_valid_type) => AnnouncementAcceptance::Accept,
+            Err(_) => {
+                tracing::trace!(target: "net::tx::policy::relaxed",
+                    %ty,
+                    %size,
+                    %hash,
+                    "Unknown transaction type in announcement. Ignoring entry."
+                );
+                AnnouncementAcceptance::Ignore
+            }
+        }
+    }
+}
+
+/// A container for policies that govern the behaviour of the `TransactionManager`.
+///
+/// This struct bundles different policies, such as how transactions are propagated and how incoming
+/// transaction announcements are filtered.
+#[derive(Debug, Clone)]
+pub struct NetworkPolicies<P, A>
+where
+    P: TransactionPropagationPolicy,
+    A: AnnouncementFilteringPolicy,
+{
+    /// Policy determining to which peers transactions are propagated.
+    pub propagation: P,
+    /// Policy determining how incoming transaction announcements are filtered,
+    /// especially concerning transaction types.
+    pub announcement_filter: A,
+}
```

Comment:
unsure if this container type is very useful if we configure this in Transactionmanager individually via generics